core/db_classes/EE_Base_Class.class.php

```diff
@@ -732,1484 +631,1729 @@ protected function _prepare_datetime_field(
+        if ($unique) {
+            $existing_extra_meta = EEM_Extra_Meta::instance()->get_one(
+                [
+                    [
+                        'EXM_key'  => $meta_key,
+                        'OBJ_ID'   => $this->ID(),
+                        'EXM_type' => $this->_model->get_this_model_name(),
+                    ],
+                ]
+            );
+            if ($existing_extra_meta) {
+                return false;
+            }
+        }
+        $new_extra_meta = EE_Extra_Meta::new_instance(
+            [
+                'EXM_key'   => $meta_key,
+                'EXM_value' => $meta_value,
+                'OBJ_ID'    => $this->ID(),
+                'EXM_type'  => $this->_model->get_this_model_name(),
+            ]
+        );
+        $new_extra_meta->save();
+        return true;
     }
 
 
     /**
-     * @param string $field_name
-     * @param string $date_format
-     * @param string $time_format
+     * Fetches a single EE_Base_Class on that relation. (If the relation is of type
+     * BelongsTo, it will only ever have 1 object. However, other relations could have an array of objects)
+     *
+     * @param string $relationName
+     * @return EE_Base_Class[] NOT necessarily indexed by primary keys
      * @throws InvalidArgumentException
      * @throws InvalidInterfaceException
      * @throws InvalidDataTypeException
      * @throws EE_Error
+     * @throws ReflectionException
      */
-    public function e_datetime($field_name, $date_format = '', $time_format = '')
+    public function get_all_from_cache($relationName)
     {
-        $this->_get_datetime($field_name, $date_format, $time_format, null, true);
+        $objects = isset($this->_model_relations[ $relationName ]) ? $this->_model_relations[ $relationName ] : [];
+        // if the result is not an array, but exists, make it an array
+        $objects = is_array($objects) ? $objects : [$objects];
+        // bugfix for https://events.codebasehq.com/projects/event-espresso/tickets/7143
+        // basically, if this model object was stored in the session, and these cached model objects
+        // already have IDs, let's make sure they're in their model's entity mapper
+        // otherwise we will have duplicates next time we call
+        // EE_Registry::instance()->load_model( $relationName )->get_one_by_ID( $result->ID() );
+        $related_model = EE_Registry::instance()->load_model($relationName);
+        foreach ($objects as $model_object) {
+            if ($related_model instanceof EEM_Base && $model_object instanceof EE_Base_Class) {
+                // ensure its in the map if it has an ID; otherwise it will be added to the map when its saved
+                if ($model_object->ID()) {
+                    $related_model->add_to_entity_map($model_object);
+                }
+            } else {
+                throw new EE_Error(
+                    sprintf(
+                        esc_html__(
```

Comment:
no... 
the translator has to have the freedom to rearrange the placeholders as needed, then sprintf swaps them for the runtime values.

if it was the other way around, then the translations would break because `esc_html__()` would get passed the runtime value, therefore making it impossible to find the string that needs to be translated.

ie: it's the difference between using `Event Name: %1$S` as an array key vs  `Event Name: My Event #1234`
the latter would never show up in the array of translatable strings.